backend/mr-db-sql/src/test/groovy/io/featurehub/db/services/FeatureUpdatesPublishingSpec.groovy

```diff
@@ -0,0 +1,143 @@
+import io.featurehub.db.api.Opts
+import io.featurehub.db.api.PersonFeaturePermission
+import io.featurehub.db.api.RolloutStrategyValidator
+import io.featurehub.mr.events.common.CacheSource
+import io.featurehub.mr.events.common.FeatureMessagingCloudEventPublisher
+import io.featurehub.mr.events.common.converter.FeatureMessagingParameter
+import io.featurehub.mr.model.Application
+import io.featurehub.mr.model.Environment
+import io.featurehub.mr.model.Feature
+import io.featurehub.mr.model.FeatureValueType
+import io.featurehub.mr.model.Portfolio
+import io.featurehub.mr.model.RoleType
+
+class FeatureUpdatesPublishingSpec extends Base2Spec {
+  PortfolioSqlApi portfolioSqlApi
+  Portfolio p1
+  ApplicationSqlApi applicationSqlApi
+  Application app
+  CacheSource cacheSource
+//  Feature feature
+  FeatureSqlApi featureSqlApi
+  EnvironmentSqlApi environmentSqlApi
+  Environment env
+  RolloutStrategyValidator rsValidator
+  PersonFeaturePermission perms
+  RolloutStrategyValidator rsv
+  FeatureMessagingCloudEventPublisher featureMessagingCloudEventPublisher
+  ApplicationSqlApi appApi
+  UUID envIdApp1
+  UUID appId
+
+  def setup() {
+    perms = new PersonFeaturePermission(superPerson, [RoleType.CHANGE_VALUE, RoleType.UNLOCK] as Set<RoleType>)
+    cacheSource = Mock()
+    rsValidator = Mock()
+    rsValidator.validateStrategies(_, _, _) >> new RolloutStrategyValidator.ValidationFailure()
+    rsValidator.validateStrategies(_, _, _, _) >> new RolloutStrategyValidator.ValidationFailure()
+    ThreadLocalConfigurationSource.createContext(['auditing.enable': 'true'])
+
+    rsv = Mock(RolloutStrategyValidator)
+    rsv.validateStrategies(_, _, _) >> new RolloutStrategyValidator.ValidationFailure()
+    featureMessagingCloudEventPublisher = Mock()
+    featureSqlApi = new FeatureSqlApi(db, convertUtils, Mock(CacheSource), rsv, featureMessagingCloudEventPublisher)
+    portfolioSqlApi = new PortfolioSqlApi(db, convertUtils, archiveStrategy)
+
+    p1 = portfolioSqlApi.getPortfolio("basic")
+
+    if (p1 == null) {
+      p1 = portfolioSqlApi.createPortfolio(new Portfolio().name("basic").description("basic"), Opts.empty(), superPerson)
+    }
+
+    applicationSqlApi = new ApplicationSqlApi(db, convertUtils, cacheSource, archiveStrategy, featureSqlApi)
+    app = applicationSqlApi.getApplication(p1.id, "app1")
+
+    if (app == null) {
+      app = applicationSqlApi.createApplication(p1.id, new Application().name("app1").description("desc1"), superPerson)
+    }
+    appId = app.id
+
+    db.currentTransaction().commit()
+
+    environmentSqlApi = new EnvironmentSqlApi(db, convertUtils, cacheSource, archiveStrategy)
+    env = environmentSqlApi.getEnvironment(app.id, "dev")
+
+    if (env == null) {
+      env = environmentSqlApi.create(new  Environment().name("dev").description("dev"), app, superPerson)
+    }
+    envIdApp1 = env.id
+    appApi = new ApplicationSqlApi(db, convertUtils, Mock(CacheSource), archiveStrategy, featureSqlApi)
```

Comment:
is it possible to refactor these duplicate impl names - e.g. appApi and applicationSqlApi and give some indication as to why you need two here?